decayedlog.go

```diff
@@ -298,23 +298,21 @@ func hashSharedSecret(sharedSecret *Hash256) HashPrefix {
 
 // Delete removes a <shared secret hash, CLTV> key-pair from the
 // sharedHashBucket.
-func (d *DecayedLog) Delete(hash []byte) error {
+func (d *DecayedLog) Delete(hash *HashPrefix) error {
 	return d.db.Batch(func(tx *bolt.Tx) error {
 		sharedHashes := tx.Bucket(sharedHashBucket)
 		if sharedHashes == nil {
 			return ErrDecayedLogCorrupted
 		}
 
-		return sharedHashes.Delete(hash)
+		return sharedHashes.Delete(hash[:])
 	})
 }
 
 // Get retrieves the CLTV of a processed HTLC given the first 20 bytes of the
 // Sha-256 hash of the shared secret.
-func (d *DecayedLog) Get(hash []byte) (uint32, error) {
-	// math.MaxUint32 is returned when Get did not retrieve a value.
```

Comment:
Why revert the default CLTV value?

Reply:
I found the behavior weird. The interface is documented as saying the the uint32 values are intentionally of ambiguous meaning for generality, yet the behavior of this method is defined assuming they are used as a CLTV. So, instead I modified the method to return an ErrEntryNotFound error instead of a MaxUint32 with no error.